# ACFLY通用配置

# 节点:
startup_px4_usb_quirk: true

# --- 系统插件 --- #

# sys_status和sys_time的连接选项
conn:
  heartbeat_rate: 1.0     # 发送心跳频率(hz)
  timeout: 10.0           # 心跳超时(s)
  timesync_rate: 5.0      # 时间同步频率(hz)，如果设置为0则禁用
  system_time_rate: 1.0   # 发送系统时间给飞控的频率

# sys_status
sys:
  min_voltage: 10.0       # 诊断的最低电压
  disable_diag: false     # 禁用所有诊断，除了心跳

# sys_time
time:
  time_ref_source: "fcu"  # 参考时间源
  timesync_mode: MAVLINK  # 时间同步模式，为同步飞控与电脑的模式
  timesync_avg_alpha: 0.6 # 时间同步平均系数
  convergence_window: 500 # 收敛滑窗大小，可大致认为收敛时间=滑窗/时间同步频率

# --- mavros插件(首字母顺序) --- #

# acfly_position_sensor
acfly_position_sensor:
  sensor_id: "camera"                       # 传感器TF名(插件将其与body_id之间建立静态tf)
  body_id: "base_link"                    # 飞控body系(通常不需要修改)
  sensor_body_rotation: [0.0, 0.0, 0.0]     # 传感器与body之间的旋转，以欧拉角rpy表示(ENU-FLU)
  sensor_body_translation: [0.0, 0.0, 0.0]  # 传感器与body之间的位移xyz(ENU-FLU)
  sensor:
    index: 15                               # 飞控注册定位传感器索引号(C9最高为16路)
    delay: 0.05                             # 默认传感器延时(s)
    trust_xy: 0.01                          # 传感器xy方向方差(m^2)
    trust_z: 0.01                           # 传感器z方向方差(m^2)
  tf:
    listen: false                           # 通过监听tf来获取位置信息(false则通过订阅来获取tf信息)
    frame_id: "map"                         # 传感器建图坐标系(通常为map)
    child_frame_id: "base_link"           # 飞控body系(通常不需要修改)
    rate_limit: 10.0                        # 监听频率

# command
cmd:
  use_comp_id_system_control: false # 有些老飞控的选项

# global_position
global_position:
  rot_covariance: 99999.0   # 角度协方差
  gps_uere: 1.0             # GPS的用户等效测距误差(m)
  use_relative_alt: true    # 对于局部坐标使用相对高度
  tf:
    send: false

# imu
imu:
  # 与IMU实际传感器参数有关，下面为MPU6000的实际值
  linear_acceleration_stdev: 0.0003
  angular_velocity_stdev: 0.0003490659 # 0.02 degrees
  orientation_stdev: 1.0
  magnetic_stdev: 0.0

# local_position
local_position:
  tf:
    send: true

# param
# 无

# rc_io
# 无

# setpoint_accel
setpoint_accel:
  send_force: false           # 是否发送力

# setpoint_raw
setpoint_raw:
  thrust_scaling: 1.0         # 推力比例因子，应该在0~1之间

# setpoint_velocity
setpoint_velocity:
  mav_frame: BODY_NED         # 合法输入为LOCAL_NED，LOCAL_OFFSET_NED，BODY_NED，BODY_OFFSET_NED

# waypoint
mission:
  pull_after_gcs: true        # 如果地面站更新，则更新任务
  use_mission_item_int: true  # 使用MISSION_ITEM_INT代替MISSION_ITEM(MISSION_ITEM_INT精度更高)

# --- mavros extra插件(首字母顺序) --- #

# play_tune
# 无

# vibration
# 无